app/reducers/app.ts

```diff
@@ -1394,7 +1395,14 @@ export const actions = {
       dispatch(actions.changeLocation(location));
       dispatch(actions.loadDirectoryContent(getLocationPath(location)));
       if (Pro && Pro.Watcher && location.watchForChanges) {
-        Pro.Watcher.watchFolder(getLocationPath(location), dispatch, actions);
+        const perspective = getCurrentDirectoryPerspective(getState());
+        const depth = perspective === perspectives.DEFAULT ? 1 : 3;
```

Comment:
I would suggest to add depth 3 only for kanban, and to make the depth parameter optional with default value 1